arviz/plots/lmplot.py

```diff
@@ -0,0 +1,331 @@
+"""Plot linear regression figure."""
+import warnings
+from numbers import Integral
+
+import xarray as xr
+import numpy as np
+from xarray.core.dataarray import DataArray
+
+from ..sel_utils import xarray_var_iter
+from ..rcparams import rcParams
+from .plot_utils import default_grid, filter_plotters_list, get_plotting_function
+
+
+def plot_lm(
+    y,
+    idata=None,
+    x=None,
+    y_model=None,
+    y_hat=None,
+    num_pp_samples=50,
+    kind_pp="samples",
+    kind_model="lines",
+    xjitter=False,
+    plot_dim=None,
+    backend=None,
+    y_kwargs=None,
+    y_hat_plot_kwargs=None,
+    y_hat_fill_kwargs=None,
+    y_model_plot_kwargs=None,
+    y_model_fill_kwargs=None,
+    backend_kwargs=None,
+    show=None,
+    figsize=None,
+    textsize=None,
+    axes=None,
+    legend=True,
+    grid=True,
+):
+    """Plot linear regression figure to visualize uncertaininty in mean and ppc values.
+
+    Parameters
+    ----------
+    y : str or DataArray or ndarray
+        If str, variable name from observed_data
+    idata : az.InferenceData object, Optional
+        Optional only if y is Sequence
+    x : str or tuple of strings or DataArray or array-like, Optional
+        If str or tuple, variable name from constant_data
+        If ndarray, could be 1D, or 2D for multiple plots
+        If none, coords name of y (y should be DataArray).
+    y_model : str or Sequence, Optional
+        If str, variable name from posterior.
+        Its dimensions should be same as y plus added chains and draws.
+    y_hat : str, Optional
+        If str, variable name from posterior_predictive.
+        Its dimensions should be same as y plus added chains and draws.
+    num_pp_samples : int, Optional, Default 50
+    kind_pp : ["samples", "hdi"], Optional, Default "samples"
```

Comment:
for arguments taking only a set of known values, the convention is to use `{` instead of `[`. Moreover, adding `default "value"` already indicated that this is an optional argument in numpydoc:

```suggestion
    kind_pp : {"samples", "hdi"}, default "samples"
```